footprints/main/models.py

```diff
@@ -22,11 +24,64 @@
 class ExtendedDateFormat(models.Model):
     edtf_format = models.CharField(max_length=256)
 
+    month_names = {
+        1: 'January', 2: 'February', 3: 'Mar', 4: 'April',
+        5: 'May', 6: 'June', 7: 'July', 8: 'August', 9: 'September',
+        10: 'October', 11: 'November', 12: 'December'}
+
     class Meta:
         verbose_name = 'Extended Date Format'
 
     def __unicode__(self):
-        return self.edtf_format
+        e = EDTF(self.edtf_format)
+
+        if e.is_interval:
+            return "%s - %s" % (self.fmt(e.date_obj.start),
+                                self.fmt(e.date_obj.end))
+        else:
+            return self.fmt(e.date_obj)
+
+    def fmt_modifier(self, date_obj):
+        if date_obj == 'open':
+            return 'present'
+        if date_obj == 'unknown':
+            return '?'
+
+    def fmt(self, date_obj):
+        result = ''
+
+        if isinstance(date_obj, basestring):
```

Comment:
I feel like it's a good practice independent of language. It reduces the nesting of conditionals and makes the code easier to read.

http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

http://www.informit.com/articles/article.aspx?p=1149121&seqNum=11